src/main/java/org/openstreetmap/atlas/generator/tools/streaming/resource/HDFSWalker.java

```diff
@@ -69,24 +118,37 @@ protected FileStatus computeNext()
                 return endOfData();
             }
 
-            final FileStatus returnValue = this.currentPaths.remove();
-            if (returnValue.isDirectory())
+            final FileStatusAtDepth returnValue = this.currentPaths.remove();
+            final FileStatus currentFileStatus = returnValue.getFileStatus();
+            final int currentFileDepth = returnValue.getFileDepth();
+
+            if (currentFileStatus.isDirectory())
             {
-                try
+                final int childDepth = currentFileDepth + ONE;
+                // Add the children only if they are below the threshold depth
+                // or the threshold is not specified at all
+                if (this.maxDepth == HDFSWalker.WALK_ALL || childDepth <= this.maxDepth)
                 {
-                    Stream.of(this.fileSystem.listStatus(returnValue.getPath()))
-                            .forEach(status -> this.currentPaths.add(status));
-                }
-                catch (final IOException oops)
-                {
-                    throw new CoreException("Can't locate children of {}", returnValue, oops);
+                    try
+                    {
+                        Stream.of(this.fileSystem.listStatus(currentFileStatus.getPath()))
+                                .forEach(status -> this.currentPaths
+                                        .add(new FileStatusAtDepth(status, childDepth)));
+                    }
+                    catch (final IOException oops)
+                    {
+                        throw new CoreException("Can't locate children of {}", currentFileStatus,
```

Comment:
Do we want to log the depth too? Something like, "... children of {} at depth {}"

Reply:
I can add it If you think that's helpful